authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerHelper.java

```diff
@@ -278,9 +278,18 @@ protected RangerRole createRangerRoleIfNotExists(String roleName, boolean isOwne
     try {
       rangerRole = rangerClient.getRole(roleName, rangerAdminName, rangerServiceName);
     } catch (RangerServiceException e) {
-      // ignore exception, If the role does not exist, then create it.
-      LOG.warn("The role({}) does not exist in the Ranger!", roleName);
+
+      // The client will return a error message contains `doesn't have permission` if the role does
+      // not exist, then create it.
+      if (e.getMessage().contains("User doesn't have permissions to get details")) {
+        LOG.warn("The role({}) does not exist in the Ranger!, e: {}", roleName, e);
```

Comment:
IIUC, this method ensures that that a specific ranger role exists.
If the role is not there yet, we'll create it.
This means we may want to print an info log rather than a warning, right?

If the user cannot check if a role exists or not, we still allow the function to continue. Does this mean that a user can create a role but he/she cannot view it?
The role becomes Write-Only for him/her?

Reply:
1. Yes, we should use info log.

2. Usually we use admin user to do these operations, so this case shouldn't occur.

Reply:
The `e.getMessage().contains("User doesn't have permissions to get details")` is Not a stable way. I think we can use another way to double check it.

Reply:
> The `e.getMessage().contains("User doesn't have permissions to get details")` is Not a stable way. I think we can use another way to double check it.

Good point. Maybe we need to treat this seriously. For whatever permission related errors, always raise a special exception type.

Reply:
I have no better way to check this. Do you have suggestion?